src/main/java/uk/nhs/hee/tis/trainee/forms/model/AbstractAuditedForm.java

```diff
@@ -129,18 +129,21 @@ public void setLifecycleState(LifecycleState lifecycleState, StatusDetail detail
         .revision(revision)
         .build();
 
-    updateStatusInfo(statusInfo);
+    updateStatusInfo(statusInfo, true);
   }
 
   /**
    * Update the form's current and historical status to include in given {@link StatusInfo}.
    *
-   * @param statusInfo The status info to add to the form.
+   * @param statusInfo           The status info to add to the form.
+   * @param affectsSubmittedDate Whether this status change affects the submitted date. In
+   *                             particular, changing the assigned admin does not affect the
+   *                             submitted date.
    */
-  private void updateStatusInfo(StatusInfo statusInfo) {
+  private void updateStatusInfo(StatusInfo statusInfo, boolean affectsSubmittedDate) {
```

Comment:
This is a fairly clumsy fix, but since this function is only used in two places it seems okish for now. A more opinionated fix might rather check if the status has actually changed, and only update `submitted` (if applicable) in that case. That would require confirmation that business rules support this.